CI: pin rust to 1.61.0 and check MSRV

the-mikedavis opened the pull request:
Pinning to a particular version brings some extra burden that we have to remember to update rust, but I think it's worth it to make CI more reproducible around new rust releases. The CI uses new rust versions almost immediately after release which means that CI may start failing faster than we can fix the lints, as in #2514. Ideally we update rust and fix clippy lints all in one go.

Connects #2528

the-mikedavis commented:
The CI rules will need to be updated to expect "Build / Test Suite (runner)" checks rather than  "Build / Test Suite (runner, rust-channel)"

pickfire commented on .github/workflows/build.yml:
```diff
@@ -16,10 +16,9 @@ jobs:
 
       - name: Install stable toolchain
-        uses: actions-rs/toolchain@v1
+        uses: helix-editor/rust-toolchain@v1
```
Why do we need our own fork?

the-mikedavis replied:
We merged in https://github.com/actions-rs/toolchain/pull/209 - without it we would need to use the deprecated `rust-toolchain` file rather than `rust-toolchain.toml`.

Looks like upstream might be unmaintained at the moment as well but we'll leave https://github.com/helix-editor/helix/issues/2528 open to switch back if 209 gets merged upstream.

pickfire commented on .github/workflows/build.yml:
```diff
@@ -112,10 +109,9 @@ jobs:
+        uses: helix-editor/rust-toolchain@v1
         with:
           profile: minimal
-          toolchain: stable
```
Why not run with both stable and the fixed version? That we can make sure helix code can still run on stable in case someone use it.

the-mikedavis replied:
Oh yeah we could have smoke tests on stable running on a cron or something. I just don't want it to fail CI for PRs when rust does a release.

the-mikedavis commented:
I updated this branch to include a matrix for checking on the stable version and the MSRV and added the fix that brings us back down from 1.60.0 to 1.57.0.
